modules/addefendBidAdapter.js

```diff
@@ -0,0 +1,79 @@
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+
+const BIDDER_CODE = 'addefend';
+
+export const spec = {
+  code: BIDDER_CODE,
+  hostname: 'https://addefend-platform.com',
+
+  getHostname() {
+    return this.hostname;
+  },
+  isBidRequestValid: function(bid) {
+    return (bid.sizes !== undefined && bid.bidId !== undefined && bid.params !== undefined &&
+              (bid.params.pageId !== undefined && (typeof bid.params.pageId === 'string')) &&
+              (bid.params.placementId !== undefined && (typeof bid.params.placementId === 'string')));
+  },
+  buildRequests: function(validBidRequests, bidderRequest) {
+    let bid = {
+      v: $$PREBID_GLOBAL$$.version,
+      auctionId: false,
+      pageId: false,
```

Comment:
are these supposed to be bools?

Reply:
yes, for the theoretical case that there are no validBidRequests (length==0) the resulting request would send false for auctionId and pageId.